src/main/kotlin/com/terraformation/backend/accelerator/ParticipantProjectSpeciesService.kt

```diff
@@ -190,11 +195,38 @@ class ParticipantProjectSpeciesService(
         FileMetadata.of(MediaType.valueOf("text/csv").toString(), filename, stream.size().toLong())
 
     fileService.storeFile("species-list-deliverable", inputStream, metadata, null) { fileId ->
-      submissionSnapshotsDao.insert(
-          SubmissionSnapshotsRow(submissionId = event.submissionId, fileId = fileId))
+      with(SUBMISSION_SNAPSHOTS) {
+        dslContext
+            .insertInto(this)
+            .set(SUBMISSION_ID, event.submissionId)
+            .set(FILE_ID, fileId)
+            .execute()
+      }
     }
   }
 
+  fun readSubmissionSnapshotFile(
+      projectId: ProjectId,
+      deliverableId: DeliverableId
+  ): SizedInputStream {
+    val submissionRecord =
+        with(SUBMISSIONS) {
+          dslContext.fetchOne(this, PROJECT_ID.eq(projectId).and(DELIVERABLE_ID.eq(deliverableId)))
+              ?: throw NotFoundException()
+        }
+
+    val submissionId = submissionRecord.id!!
+
+    requirePermissions { readSubmission(submissionId) }
+
+    // Make sure the file belongs to the submission
```

Comment:
Is this a to-do item?

Reply:
No, just poorly worded. What it should say is "get the snapshot that belongs to the submission" which has the file ID that belongs to the submission.